src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/SpeculationAnalyzer.cs

```diff
@@ -773,6 +773,15 @@ bool IsSupportedConstructWithNullType()
                 return true;
             }
 
+            // Similar to above, it's fine for a tuple expression to have a 'null' direct type (as long as a
+            // target-typed tuple-expression conversion happened).  Note: unlike above, we don't have to check
+            // a language version since tuple expressions always supported tuple-expression-conversions.
+            if (newExpression.IsKind(SyntaxKind.TupleExpression) &&
+                this.SpeculativeSemanticModel.GetConversion(newExpression).IsTupleLiteralConversion)
+            {
+                return true;
```

Comment:
Is there any extra condition needed here? Like:

```c#
SymbolsAreCompatible(originalTypeInfo.Type, newTypeInfo.ConvertedType)
```

I see the tests are working too if having this extra condition

Reply:
Yes. The types should be validated as well.

Reply:
Added